src/Microsoft.DotNet.Cli.Utils/PathUtility.cs

```diff
@@ -88,6 +88,34 @@ public static void EnsureDirectoryExists(string directoryPath)
             }
         }
 
+        public static bool TryEnsureDirectoryExists(string directoryPath)
+        {
+            try
+            {
+                EnsureDirectoryExists(directoryPath);
+
+                return true;
+            }
+            catch
```

Comment:
I'm not saying it's not valid to swallow some exceptions, just that it's probably not valid to swallow all exceptions. We should specify which ones we want to swallow, otherwise we might hide issues like the ones I outlined.